content/posts/2021/reana-client-python-api.md

```diff
@@ -0,0 +1,170 @@
+    }  # Parameters for your workflow
+}
+```
+
+- Then we will provide the actual workflow specification,
+  in this case using serial workflow:
+  (see the [reana.yaml](https://github.com/reanahub/reana-demo-root6-roofit/blob/f29e98b482fe8cb801735ac2fa48bc01e6cc05b7/reana.yaml#L13-L22)
+  code that corresponds to it
+  )
+```python
+workflow_json = {
+    'steps': [
+        {'name': 'gendata',
+         'environment': 'reanahub/reana-env-root6:6.18.04',
+         'commands': [
+           'mkdir -p results',
+           'root -b -q \'code/gendata.C(${events},"${data}")\' | tee gendata.log']},
+        {'name': 'fitdata',
+         'environment': 'reanahub/reana-env-root6:6.18.04',
+         'commands': [
+           'root -b -q \'code/fitdata.C("${data}","${plot}")\' | tee fitdata.log']}]
+}
+```
+
+We are now ready to create our workflow in REANA:
+```python
+# with `reana-client` it would be
+#   `reana-client create -n $my_workflow_name`
+from reana_client.api.client import create_workflow_from_json
+create_workflow_from_json(
+    workflow_json=workflow_json,
+    name=my_workflow_name,
+    access_token=my_reana_token,
+    parameters=my_inputs,
+    workflow_engine=workflow_type)
+```
+
+Upload files to the workflow workspace:
+```python
+# with `reana-client` it would be
+#   `reana-client upload -w $my_workflow_name`
+from reana_client.api.client import upload_to_server
+
+abs_path_to_input_files = \
+  [os.path.abspath(f) for f in my_inputs['files']]
+upload_to_server(my_workflow_name,
+                 abs_path_to_input_files,
+                 my_reana_token)
+```
+
+And now we can start our workflow
+```python
+# with `reana-client` it would be
+#   `reana-client start -w $my_workflow_name`
+from reana_client.api.client import start_workflow
+
+start_workflow(my_workflow_name, my_reana_token, {})
+```
+
+We can inquiry for it's status as follows:
+```python
+# with `reana-client` it would be
+#   `reana-client status -w $my_workflow_name`
+import time
+
+while True:
+  from reana_client.api.client import get_workflow_status
+  status_details = get_workflow_status(my_workflow_name,
+                                      my_reana_token)
+  if status_details['status'] == 'finished':
+    break
+  print('Current status: ', status_details['status'])
+  time.sleep(1)
+```
+
+Once finished, we can list the output files and download
+the ones we are interested in:
+
+- Listing files:
+```python
+# with `reana-client` it would be
+#   `reana-client ls -w $my_workflow_name`
+from reana_client.api.client import list_files
+
+list_files(my_workflow_name, my_reana_token)
+```
+
+- Downloading the result file:
+```python
+# with `reana-client` it would be
+#   `reana-client download results/plot.png -w $my_workflow_name`
+from reana_client.api.client import download_file
+
+output_filename = 'results/plot.png'
+file_binary_blob = download_file(
+  my_workflow_name, output_filename, my_reana_token)
+```
+
+Here you can see the final result:
```

Comment:
Add instructions how to plot it, perhaps from Python, perhaps just `os.system()` with `xdg-open`.  The motivation is to provide full instructions so that everything would be runnable.